### ISSUE TITLE
Make slow queries more pleasant within the UI

### ISSUE THREAD

**Issue by John-Gee** [theotherp/nzbhydra2#914]
Hello,

When I manually run queries against my various indexers some of them take a long time to reply.  I am not sure how you currently do it, so I apologize if I say anything dumb/redundant.

1- Since we have stats, it'd be probably nicer, on average, to make sure to start querying the slower indexers first, though if you fire all queries within a few ms already that would not make much of a difference.

2- It'd be also nice to have a soft timeout limit, in the sense that after that timeout queries will continue to run but in the background so we don't have to wait to start interacting. Of course that would be useless if no result have been found yet and I'm not certain how to properly update the UI without disturbing the user. Likely this would be an opt-in feature.

Thank you!

**Comment by theotherp** [comment]
1) It's indeed a matter of 1 or 2 ms as they're queried in multiple threads in parallel. The order is irrelevant in this case.

2) Adding additional results on the go while more indexers respond would result in unpredictable changes to the table. I could (and already considered a long time ago) replacing the "Cancel" button with a kinda of "That's enough" functionality where you're shown the results from the indexers which have already responded. There's really no reason to have Cancel abort everything. But as I've said in other posts I really don't have much free time right now. I'll keep it in mind.

**Comment by John-Gee** [comment]
Yeah the cancel does indeed not make much sense I agree.

**Comment by theotherp** [comment]
How do you run hydra? I could send you a beta version but only if you're not running docker.

**Comment by John-Gee** [comment]
I run the compiled version on straight Linux x64.

I've thought a bit more about "Adding additional results on the go", it could be part of the "Load More" button, or a separate one if that makes more sense.

Thank you!

**Comment by theotherp** [comment]
Try this then: https://ufile.io/12grcpa9

Just replace the file in the main folder.

**Comment by John-Gee** [comment]
With that version the cancel button does not seem to do anything. I keep clicking it but nothing happens.

**Comment by theotherp** [comment]
You'r right. WTF, it worked yesterday...

**Comment by theotherp** [comment]
Got it, the UI source changes got reverted by mistake. This one should work: https://ufile.io/cnlygr8x

**Comment by John-Gee** [comment]
Yup that works! Thank you!
